### PR TITLE
Feat: improve metrics reporting

### PR DESCRIPTION
**Pull request by jdeschamps**
## Disclaimer

- [ ] I am an AI agent.
- [x] I have used AI and I thoroughly reviewed every line.
- [ ] I have not used AI extensively.

## Description

> [!NOTE]  
> **tldr**: Improve reporting from reading `CSVLogger` by structuring the data and adding metrics.

`CAREamist` allows reading the losses from the `CSVLogger`, returning a dictionary. However, this dictionary does not include the metrics, which may differ from algorithm to algorithm. 

This PR improves the data reading from the csv file, structuring the various elements as `Series` a dataclass that can easily be plotted and the return type of `read_csv_logger` as a `TrainingReport`:

```python
import matplotlib.pyplot as plt

report = read_csv_logger(...)
train_loss = report.train_loss
plt.plot(train_loss.epoch, train_loss.value)

# metrics
print(report.metrics.keys())

# then plot whichever metrics
```

```python
import matplotlib.pyplot as plt
history = careamist.get_losses()
plt.plot(history.train_loss.epoch, history.train_loss.value, label="train")
plt.plot(history.val_loss.epoch, history.val_loss.value, label="val")
plt.xlabel("Epoch")
plt.ylabel("Loss")
plt.legend()
```

Following #1043, this PR updated the `plot_loss` function:

```python
from careamics.plotting import plot_loss

training_report = careamist.get_losses()
plot_loss(training_report) 
```

plot_loss

---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [x] New tests have been added (for bug fixes/features)
- [x] Documentation has been updated
- [x] Pre-commit passes

### CODE REVIEW

**Comment by jdeschamps** [comment]
Docs needs to be updated.

[EDIT]: Docs updated

**Comment by jdeschamps** [comment]
Need to update changes from #1043 

[EDIT]: done in https://github.com/CAREamics/careamics/pull/1058/commits/b3e73b509af71bde955f728fd34046f3edd691ee

File: src/careamics/plotting/loss.py [hunk 1]
```diff
@@ -3,50 +3,106 @@
...
+        ax[plot_idx].set_xlabel("Epoch")
+        ax[plot_idx].set_ylabel("Value")
+        ax[plot_idx].set_title("Learning rate")
+        ax[plot_idx].xaxis.set_major_locator(MaxNLocator(integer=True))
+        ax[plot_idx].legend(frameon=False)
+
+    fig.tight_layout()
 
     # check for saving the figure as an image
     if save_path is not None:
```
Comment (mese79) [inline src/careamics/plotting/loss.py:108]: All seems good :)  
Although another way to handle plot axes could be like:
```python
fig = plt.figure(1, figsize=fig_size, layout="constrained")

# loss
ax = fig.add_subplot(1, n_plots, plot_index)
...

# metrics
if plot_metrics:
    plot_idx += 1
    ax = fig.add_subplot(1, n_plots, plot_index)
    ...

# lr
if plot_learning_rate:
    plot_idx += 1
    ax = fig.add_subplot(1, n_plots, plot_index)
    ...

**Review by mese79 — APPROVED** [review]

**Comment by mese79** [comment]
And the `tests/plotting/test_plotting.py` should get updated as well!

File: src/careamics/plotting/loss.py (hunk 1 above)
Comment (jdeschamps) [reply src/careamics/plotting/loss.py:104]: Done in https://github.com/CAREamics/careamics/pull/1058/changes/bafcf4cb15017279c294de961e4a5672be23394d

**Comment by jdeschamps** [comment]
> And the `tests/plotting/test_plotting.py` should get updated as well!

Ah good point, done in https://github.com/CAREamics/careamics/pull/1058/changes/000cc28c6565a1f3c23379698ec979b0c1b921e5
